src/python/WMCore/MicroService/Unified/Transferor.py

```diff
@@ -72,88 +86,290 @@ def close(self):
         "Close backend store"
         self.db.close()
 
-def checkRequests(func, store, interval):
-    "Demon to perform given function action for all request in our store"
+def daemon(func, req_status, interval):
+    "Daemon to perform given function action for all request in our store"
     while True:
         try:
-            requests = store.info().keys()
-        except ValueError: # store is closed
-            requests = []
-        for request in requests:
-            func(request)
+            func(req_status)
+        except Exception as exc:
+            pass
         time.sleep(interval)
 
-class RequestManager(object):
+class MSManager(object):
     "Class to keep track of transfer progress in PhEDEx for a given task"
-    def __init__(self, group='DataOps', dbFileName=None, interval=10, verbose=False):
+    def __init__(self, svc, group='DataOps', dbFileName=None, interval=60, logger=None, verbose=False):
+        if logger:
+            self.logger = logger
+        else:
+            self.logger = logging.getLogger('reqmgr2ms:MSManager')
+            logging.basicConfig()
         self.verbose = verbose
+        if verbose:
+            self.logger.setLevel(logging.DEBUG)
         if not dbFileName:
             fobj = tempfile.NamedTemporaryFile()
             dbFileName = '%s.db' % fobj.name
         self.store = RequestStore(dbFileName)
-        self.phedex = PhEDEx()
+        self.logger.debug("dbFileName %s" % dbFileName)
+        self.phedex = PhEDEx() # eventually will change to Rucio
         self.group = group
-        thname = 'RequestManager monitor'
-        self.thr = start_new_thread(thname, checkRequests, \
-                (self.checkStatus, self.store, interval))
-        if verbose:
-            print("### Running %s thread, running: %s" % (thname, self.thr.running()))
-        print("RequestManager, group=%s, db=%s, interval=%s" % (group, dbFileName, interval))
+        self.svc = svc # Services: ReqMgr, ReqMgrAux
+        thname = 'MSTransferor'
+        self.thr = start_new_thread(thname, daemon, (self.transferor, 'assigned', interval))
+        self.logger.debug("### Running %s thread %s" % (thname, self.thr.running()))
+        thname = 'MSTransferorMonit'
+        self.ms_monit = start_new_thread(thname, daemon, (self.monit, 'staging', interval))
+        self.logger.debug("+++ Running %s thread %s" % (thname, self.ms_monit.running()))
+        self.logger.info("MSManager, group=%s, db=%s, interval=%s" % (group, dbFileName, interval))
+
+    def monit(self, req_status='staging'):
+        """
+        MSManager monitoring function.
+        It performs transfer requests from staging to staged state of ReqMgr2.
+        For references see
+        https://github.com/dmwm/WMCore/wiki/ReqMgr2-MicroService-Transferor
+        """
+        try:
+            # get requests from ReqMgr2 data-service for given statue
+            requestSpecs = self.svc.reqmgr.getRequestByStatus([req_status], detail=True)
```

Comment:
We don't need request details here (detail=False). We only need to know workflow names in this status, all the rest of the information should be available from the application database containinig a map of workflow -- dataset -- transfer id

Reply:
ok